File: problem268.cs
public class Solution {
    public int MissingNumber(int[] nums) {
        HashSet<int> set = new HashSet<int>();
        foreach (int num in nums) {
            set.Add(num);
        }

        int i;
        for (i = 0; i <= nums.Length; i++) {
            if (!set.Contains(i)) {
                return i;
            }
        }
        return i;
    }
}

File: su19/problem268.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        actual = nums[0]
        expected = 0
        for i in range(1, len(nums)):
            actual ^= nums[i]
            expected ^= i
        expected ^= len(nums)
        missing = actual ^ expected
        return missing